Component generator doesn't fully handle mixed casing

I generated a new Redwood component using `yarn rw g component NFTImage`. As expected, it created a bunch of files in `web/src/components/NftImage`, here's the output:
```
yarn run v1.22.11
$ /Users/m/projects/ens-gallery/app/node_modules/.bin/rw g component NFTImage
  ✔ Generating component files...
    ✔ Successfully wrote file `./web/src/components/NftImage/NftImage.test.tsx`
    ✔ Successfully wrote file `./web/src/components/NftImage/NftImage.stories.tsx`
    ✔ Successfully wrote file `./web/src/components/NftImage/NftImage.tsx`
✨  Done in 1.76s.
```

When I then ran `yarn rw storybook`, I got the following error:
```
ModuleNotFoundError: Module not found: Error: [CaseSensitivePathsPlugin] `/Users/m/projects/ens-gallery/app/web/src/components/NftImage/NFTImage.tsx` does not match the corresponding path on disk `NftImage.tsx`.
```

If I run `yarn rw g component NftImage`, it all works fine.

I'm using Redwood v0.36.4.

--- a/packages/cli/src/commands/generate/__tests__/helpers.test.js
+++ b/packages/cli/src/commands/generate/__tests__/helpers.test.js
@@ -85,7 +85,7 @@ test('customOrDefaultTemplatePath returns the app path with proper side, generat
 })
 
 test('templateForComponentFile creates a proper output path for files', () => {
-  const names = ['FooBar', 'fooBar', 'foo-bar', 'foo_bar', 'FOO_BAR']
+  const names = ['FooBar', 'fooBar', 'foo-bar', 'foo_bar', 'FOO_BAR', 'FOOBar']
 
   names.forEach((name) => {
     const output = helpers.templateForComponentFile({

--- a/packages/cli/src/lib/__tests__/index.test.js
+++ b/packages/cli/src/lib/__tests__/index.test.js
@@ -64,7 +64,7 @@ test('nameVariants returns a single word cased variables', () => {
 })
 
 test('nameVariants returns a multi word cased variables', () => {
-  const names = ['FooBar', 'fooBar', 'foo_bar', 'foo-bar']
+  const names = ['FooBar', 'fooBar', 'foo_bar', 'foo-bar', 'FOOBar']
 
   names.forEach((name) => {
     const vars = index.nameVariants(name)

--- a/packages/cli/src/lib/index.js
+++ b/packages/cli/src/lib/index.js
@@ -129,7 +129,7 @@ export const nameVariants = (name) => {
   const normalizedName = pascalcase(paramCase(pluralize.singular(name)))
 
   return {
-    pascalName: pascalcase(name),
+    pascalName: pascalcase(paramCase(name)),
     camelName: camelcase(name),
     singularPascalName: normalizedName,
     pluralPascalName: pluralize(normalizedName),

